mechanics.md

```diff
@@ -4,63 +4,65 @@ So, how do I use this?
 
 ## I Have an Idea!
 
-Ideas should have
- * a pretty specific goal: a user story ("Users can..."), dependency ("This will allow ..") or a completion condition ("All services .."); and
- * some vague idea of how this might be accomplished, or a few alternatives.
+We've learned that ideas should be developed outside of the RFC process first.
+Create a document in another context and talk it through with the relevant people.
+Keep iterating until there is one option that has rough agreement from those people.
+It's OK if some details are still unclear.
 
-If you have an idea, here's how to get started:
+A symptom of rushing this phase is an enormous list of comments on the RFC pull request.
+This is usually difficult to follow and communications issues quickly swamp the process.
+If this occurs, return to the idea phase.
+
+## I Think We Have Rough Consensus!
+
+By this time you should have:
+
+ * a pretty specific goal: a user story ("Users can..."), dependency ("This will allow ..") or a completion condition ("All services ..");
+ * a pretty solid idea of the solution; and
+ * tentative agreement from most of the people concerned.
+
+It's time to make an RFC!
 
 * Copy [`template.txt`](template.txt) to `rfcs/xxxx-<rfc-title>.md`.
-  Fill in the first few sections, but feel free to leave things TBD at this stage.
-  Commit it to a branch, push, and make a pull request.
+  Be as detailed as possible.
+  If there are open questions, answer them before proceeding.
+  Commit it to a branch, push, and make a *draft* pull request.
 
 * Once you have the pull request number, modify the filename to include it (`rfcs/<number>-<rfc-title>.md`).
   Replace `<number>` in the file with the PR number as well.
-  Push again.
+  Push again, and mark the PR as ready for review, and add the label `Phase: Proposal` to the PR.
+
+* Champion the RFC.
+
+  Get anybody you know will care about this on-board quickly by contacting them
+  by whatever means you want: irc, email, @ mentions should all work well.  As
+  early as possible, get an email out to the tools-tc list if this is going to
+  be something more than a couple specific people care about. Use your best
+  judgement.  Give a few days for opinions to filter in. Respond promptly if
+  possible to avoid a confusing conversation but give enough time for people
+  who are PTO or busy etc.
 
-You can leave the PR un-assigned, or assign it to yourself if you intend to champion it.
-Shop the issue around to the team and outside the team to try to get a diversity of opinions on it.
-It's OK to leave ideas open indefinitely, awaiting their time.
+  The idea here is to work out any disagreement and come to a proposal everyone
+  can live with, so modify the proposal as necessary using the usual git tools.
+  Reaching consensus is unlikely, but compromise is always possible.
```

Comment:
I wonder if RFCs should contain the alternatives considered and which trade-offs were chosen in the end?

Reply:
That's a really good point.  Maybe when this lands you can make a PR to add that?  There have been some good examples around the Rust async work, e.g., https://internals.rust-lang.org/t/await-syntax-discussion-summary/9914

Reply:
(I think this is a great idea, I just don't want to re-request review from everyone who's already agreed to this PR)